core/shared/src/main/scala/zio/IsSubtype.scala

```diff
@@ -0,0 +1,27 @@
+package zio
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("\nOutput Type Mismatch\n  expected: ${B}\n    actual: ${A}\n\n")
+sealed abstract class HasOutput[-A, +B] extends (A => B)
```

Comment:
I wonder if we want to avoid using the term `Has` here given its existing definition as another data type? Something like `IsSubtypeOutput` or `IsSubtypeOfOutput` might be more descriptive and reads relatively well in infix position. Like `implicit ev: A IsSubtypeOfOutput Option[B]`. We could simplify if we didn't care about distinguishing the channels but I think that is helpful to indicate to the user where something went wrong.

Reply:
That crossed my mind—Has is accounted for. I like `implicit ev: A IsSubtypeOfOutput Option[B]` a lot.